pkg/cli/common.go

```diff
@@ -44,3 +48,48 @@ func EnsureDestinationNotExist(dest string) error {
 	}
 	return nil
 }
+
+func LookupCacheHosts(srvName string) ([]string, error) {
+	_, srvs, err := net.LookupSRV("http", "tcp", srvName)
+	if err != nil {
+		return nil, err
+	}
+	return orderCacheHosts(srvs)
+}
+
+var hostnameIndexRegexp = regexp.MustCompile(`^[a-z0-9-]*-([0-9]+)[.]`)
+
+func orderCacheHosts(srvs []*net.SRV) ([]string, error) {
+	// loop through to find highest index
+	highestIndex := 0
+	for _, srv := range srvs {
+		cacheIndex, err := cacheIndexFor(srv.Target)
+		if err != nil {
+			return nil, err
+		}
+		if cacheIndex > highestIndex {
+			highestIndex = cacheIndex
+		}
+	}
+	output := make([]string, highestIndex+1)
+	for _, srv := range srvs {
+		cacheIndex, err := cacheIndexFor(srv.Target)
+		if err != nil {
+			return nil, err
+		}
+		hostname := strings.TrimSuffix(srv.Target, ".")
+		if srv.Port != 80 {
+			hostname = fmt.Sprintf("%s:%d", hostname, srv.Port)
+		}
+		output[cacheIndex] = hostname
+	}
+	return output, nil
```

Comment:
This doesn't need to be changed to land the code:

We could avoid the second pass through `srvs`, but i'm not sure if that would improve the readability of the code. I'm making this comment because it took me longer than I'd like to admit to tease out the dual loops.

Reply:
yeah it's a go thing. In another language I might have a single function to find the highest cache index but Go really wants to make it explicit when you're looping through a slice so you don't accidentally write O(n^2) code when you think you're writing O(n) code.

I could add comments to make it clearer what the purpose of each loop is?